### PR TITLE
remove unnecessary type parameter on showBottomSheet()

### PR DESCRIPTION
**Pull request by a14n**

### CODE REVIEW

File: packages/flutter/lib/src/material/scaffold.dart [hunk 1]
```diff
@@ -1228,8 +1228,8 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
     }
   }
 
-  PersistentBottomSheetController<T> _buildBottomSheet<T>(WidgetBuilder builder, AnimationController controller, bool isLocalHistoryEntry) {
-    final Completer<T> completer = Completer<T>();
+  PersistentBottomSheetController<void> _buildBottomSheet(WidgetBuilder builder, AnimationController controller, bool isLocalHistoryEntry) {
+    final Completer<void> completer = Completer<void>();
```
Comment (a14n) [inline packages/flutter/lib/src/material/scaffold.dart]: This completer only completes without value and it seems to be the only place where `T` was used.

File: packages/flutter/lib/src/material/bottom_sheet.dart [hunk 2]
```diff
@@ -319,14 +319,14 @@ class _ModalBottomSheetRoute<T> extends PopupRoute<T> {
...
-Future<T> showModalBottomSheet<T>({
+Future<void> showModalBottomSheet({
   @required BuildContext context,
   @required WidgetBuilder builder,
 }) {
   assert(context != null);
   assert(builder != null);
   assert(debugCheckHasMaterialLocalizations(context));
-  return Navigator.push(context, _ModalBottomSheetRoute<T>(
+  return Navigator.push(context, _ModalBottomSheetRoute(
```
Comment (dnfield) [inline packages/flutter/lib/src/material/bottom_sheet.dart]: Can't this potentially return something other than a `Future<void>`?

Comment (a14n) [reply packages/flutter/lib/src/material/bottom_sheet.dart]: I haven't found any place where something other than `void` (or `Null`) is returned. The only place where `T` is used is to create a `Completer<T>` which is only [complete without value](https://github.com/flutter/flutter/pull/22899/files#diff-ad7e5ee7be5375e4242652ddc6448da7R1245). So I looks fine to remove the type parameter on `_ModalBottomSheetRoute` and make it extending `PopupRoute<void>`.

File: packages/flutter/lib/src/material/bottom_sheet.dart [hunk 3]
```diff
@@ -319,14 +319,14 @@ class _ModalBottomSheetRoute<T> extends PopupRoute<T> {
 ///  * [showBottomSheet] and [ScaffoldState.showBottomSheet], for showing
 ///    non-modal bottom sheets.
 ///  * <https://material.google.com/components/bottom-sheets.html#bottom-sheets-modal-bottom-sheets>
-Future<T> showModalBottomSheet<T>({
+Future<void> showModalBottomSheet({
```
Comment (Hixie) [inline packages/flutter/lib/src/material/bottom_sheet.dart]: this is probably a breaking change, so we probably don't want to change that.

File: packages/flutter/lib/src/material/bottom_sheet.dart [hunk 4]
```diff
@@ -370,11 +370,11 @@ Future<T> showModalBottomSheet<T>({
 ///    sheet.
 ///  * [Scaffold.of], for information about how to obtain the [BuildContext].
 ///  * <https://material.google.com/components/bottom-sheets.html#bottom-sheets-persistent-bottom-sheets>
-PersistentBottomSheetController<T> showBottomSheet<T>({
+PersistentBottomSheetController<void> showBottomSheet({
```
Comment (Hixie) [inline packages/flutter/lib/src/material/bottom_sheet.dart]: ditto

File: packages/flutter/lib/src/material/bottom_sheet.dart (hunk 2 above)
Comment (Hixie) [reply packages/flutter/lib/src/material/bottom_sheet.dart]: i would expect `Navigator.pop()` to have its value used here.

File: packages/flutter/lib/src/material/bottom_sheet.dart (hunk 4 above)
Comment (Hixie) [reply packages/flutter/lib/src/material/bottom_sheet.dart]: i think with this one it's true that there's no current way to provide a return value, but that's probably just an oversight in the api. The close method ideally would take an optional T.

File: packages/flutter/lib/src/material/bottom_sheet.dart (hunk 3 above)
Comment (a14n) [reply packages/flutter/lib/src/material/bottom_sheet.dart]: It is a breaking change but I usually never see type parameters on method like this one. So I wouldn't expect too much problem.

BTW changing `Future<Null>` to `Future<void>` is also a breaking change as `Future<void>` is not assignable to `Future<Null>` (throws at runtime).

**Comment by a14n** [comment]
> that's probably just an oversight in the api. The close method ideally would take an optional T.

This PRs was based on the current codebase. So I'm going to close this PR regarding your comment.

Thanks for the review.

**Pull request closed without merging**